### PR TITLE
Testcase

### PR DESCRIPTION
**Pull request by kaimal11**

### CODE REVIEW

File: tests/query_test/qpattern_test.cpp [hunk 1]
```diff
@@ -0,0 +1,2298 @@
...
+
+    auto qpattern = query::QuadPattern {graph, sub, pred, obj};
+
+    SUBCASE("Check validity & output"){
+        CHECK(qpattern.graph() == graph);
+        CHECK(qpattern.subject() == sub);
+        CHECK(qpattern.predicate() == pred);
+        CHECK(qpattern.object() == obj);
+
+        CHECK(qpattern.valid());
```
Comment (bigerl) [inline tests/query_test/qpattern_test.cpp]: This line `CHECK(qpattern.valid());` doesn't make sense but the following two lines do. Remove?

**Review by bigerl — CHANGES_REQUESTED** [review]
In general, the PR looks very good. It's mainly formal things that need further work. 

- [x] rename dir `other_test` to `nodes`
- [x] rename dir `query_test` to `query`
- [x] rename cpp files for tests to `tests_{classname}.cpp`, e.g. `tests_TriplePattern.cpp`
- [x] one `add_executable` per `tests_{classname}.cpp`. Executable name: `tests_{classname}`
  e.g.: `add_executable(tests_TriplePattern query/tests_TriplePattern.cpp)`
- [x] use clang format on all files you created (see [.clang-format with CLion](https://www.jetbrains.com/help/clion/clangformat-as-alternative-formatter.html))
- [x] remove autogenerated comments like: `Created by kaimal on 18.11.21.`
- [x] rename `method` in the QuadPattern and TriplePattern tests to something like `validate_quad_pattern` and add a short doxygen docu for the function
- [x] replace `<http://looneytunes-graph.com#>` by `<http://example.com/>`
- [x] use numbers or letters for differnt IRIs, e.g. `<http://example.com/A>`, `<http://example.com/B>`, `<http://example.com/C>`
- [x] use more neutral strings, like `"A"`, `"B"`, `"C"` or `"str1"`, `"str3"`, etc. instead of things like `"Bugs Bunny"`
- [x] remove ` CHECK(qpattern.valid());` l. 19 `qpattern_test.cpp`

File: tests/query_test/qpattern_test.cpp (hunk 1 above)
Comment (kaimal11) [reply tests/query_test/qpattern_test.cpp]: Changed everything as per request except pt no.4 (one add_executable per tests_{classname}.cpp. Executable name: tests_{classname} e.g.: add_executable(tests_TriplePattern query/tests_TriplePattern.cpp)). Error occurring while adding executable in CMakeLists.txt

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -10,7 +10,12 @@ FetchContent_Declare(
 FetchContent_MakeAvailable(DocTest)
 
 # add the executable for all tests
-add_executable(tests_hello_test hello_test/main.cpp query_test/variable_test.cpp query_test/tpattern_test.cpp query_test/qpattern_test.cpp others_test/literal_test.cpp)
+add_executable(tests_hello_test hello_test/main.cpp query/tests_Variable.cpp query/tests_TriplePattern.cpp query/tests_QuadPattern.cpp nodes/tests_Literal.cpp)
+#[[add_executable(tests_hello_test hello_test/main.cpp)
```
Comment (bigerl) [inline tests/CMakeLists.txt]: please comment in the single executables for single cpp files and remove the executable that includes all cpp files

**Review by bigerl — CHANGES_REQUESTED** [review]

**Comment by bigerl** [comment]
Did you run clang format on the cpp source files?

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -10,7 +10,12 @@ FetchContent_Declare(
...
-add_executable(tests_hello_test hello_test/main.cpp)
+add_executable(tests_hello_test hello_test/main.cpp query/tests_Variable.cpp query/tests_TriplePattern.cpp query/tests_QuadPattern.cpp nodes/tests_Literal.cpp)
+#[[add_executable(tests_hello_test hello_test/main.cpp)
+add_executable(tests_Variable query/tests_Variable.cpp)
+add_executable(tests_TriplePattern query/tests_TriplePattern.cpp)
+add_executable(tests_QuadPattern query/tests_QuadPattern.cpp)
+add_executable(tests_Literal nodes/tests_Literal.cpp)]]
 target_link_libraries(tests_hello_test
         doctest
         rdf4cpp
```
Comment (bigerl) [inline tests/CMakeLists.txt]: you need to add those dependencies to each target (`add_executable`) for them to work. Additionally, you need to add the line 
`#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN` before `#include <doctest/doctest.h>` in the source file

**Review by bigerl — CHANGES_REQUESTED** [review]
Please use this CMakeLists.txt:
```
## C++ language configuration boilerplate

include(FetchContent)

FetchContent_Declare(
        DocTest
        GIT_REPOSITORY "https://github.com/onqtam/doctest"
        GIT_TAG  "2.4.6"
)
FetchContent_MakeAvailable(DocTest)

# add the executable for all tests
add_executable(tests_hello_test hello_test/main.cpp)
target_link_libraries(tests_hello_test
        doctest
        rdf4cpp
        )
set_property(TARGET tests_hello_test PROPERTY CXX_STANDARD 20)

add_executable(tests_Variable query/tests_Variable.cpp)
target_link_libraries(tests_Variable
        doctest
        rdf4cpp
        )
set_property(TARGET tests_Variable PROPERTY CXX_STANDARD 20)

add_executable(tests_TriplePattern query/tests_TriplePattern.cpp)
target_link_libraries(tests_TriplePattern
        doctest
        rdf4cpp
        )
set_property(TARGET tests_TriplePattern PROPERTY CXX_STANDARD 20)

add_executable(tests_QuadPattern query/tests_QuadPattern.cpp)
target_link_libraries(tests_QuadPattern
        doctest
        rdf4cpp
        )
set_property(TARGET tests_QuadPattern PROPERTY CXX_STANDARD 20)

add_executable(tests_Literal nodes/tests_Literal.cpp)
target_link_libraries(tests_Literal
        doctest
        rdf4cpp
        )
set_property(TARGET tests_Literal PROPERTY CXX_STANDARD 20)

# include(${doctest_SOURCE_DIR}/scripts/cmake/doctest.cmake)
# doctest_discover_tests(tests_hello_test)

# copy files for testing to the binary folder
# file(COPY foldername DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
```

That should do

**Comment by kaimal11** [comment]
> Please use this CMakeLists.txt:
> 
> ```
> ## C++ language configuration boilerplate
> 
> include(FetchContent)
> 
> FetchContent_Declare(
>         DocTest
>         GIT_REPOSITORY "https://github.com/onqtam/doctest"
>         GIT_TAG  "2.4.6"
> )
> FetchContent_MakeAvailable(DocTest)
> 
> # add the executable for all tests
> add_executable(tests_hello_test hello_test/main.cpp)
> target_link_libraries(tests_hello_test
>         doctest
>         rdf4cpp
>         )
> set_property(TARGET tests_hello_test PROPERTY CXX_STANDARD 20)
> 
> add_executable(tests_Variable query/tests_Variable.cpp)
> target_link_libraries(tests_Variable
>         doctest
>         rdf4cpp
>         )
> set_property(TARGET tests_Variable PROPERTY CXX_STANDARD 20)
> 
> add_executable(tests_TriplePattern query/tests_TriplePattern.cpp)
> target_link_libraries(tests_TriplePattern
>         doctest
>         rdf4cpp
>         )
> set_property(TARGET tests_TriplePattern PROPERTY CXX_STANDARD 20)
> 
> add_executable(tests_QuadPattern query/tests_QuadPattern.cpp)
> target_link_libraries(tests_QuadPattern
>         doctest
>         rdf4cpp
>         )
> set_property(TARGET tests_QuadPattern PROPERTY CXX_STANDARD 20)
> 
> add_executable(tests_Literal nodes/tests_Literal.cpp)
> target_link_libraries(tests_Literal
>         doctest
>         rdf4cpp
>         )
> set_property(TARGET tests_Literal PROPERTY CXX_STANDARD 20)
> 
> # include(${doctest_SOURCE_DIR}/scripts/cmake/doctest.cmake)
> # doctest_discover_tests(tests_hello_test)
> 
> # copy files for testing to the binary folder
> # file(COPY foldername DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
> ```
> 
> That should do

implemented the same and also reformated code to clang format

**Review by bigerl — APPROVED** [review]

**Pull request merged**